docs/examples/gmt-1155.md

```diff
@@ -9,400 +9,475 @@ sidebar_position: 6
 
-## Default multitoken implementation
-The functions that must be supported by each multi-token contract:
-- *mint(to, []token_id, []metadata, []amounts)* - is a function that creates single/multiple new tokens (with the corresponding supply from amounts array). Metadata can include any information about the token: it can be a link to a specific resource, a description of the token, etc. Metadata is stored for NFTs only;
-- *burn(from, []token_id, []amounts)* - is a function that removes the specified amounts of tokens with mentioned ids from the contract;
-- *transfer(to, []token_id, []amounts)* - is a function that allows you to transfer tokens with the token_id  to the to account;
-- *approve/revoke approval(approved account, token_id)* - is a function that allows you to give the right to dispose of the token to the specified approved_account. This functionality can be useful on marketplaces or auctions as when the owner wants to sell his token, they can put it on a marketplace/auction, so the contract will be able to send this token to the new owner at some point;
-- *balance(account)* - is a function that returns the ids and amounts of different tokens a user has in possession;
+## Architecture
 
+Because of the use of an asynchronous model and actor model, the implementation architecture requires splitting the contract into several parts, similar to sharded-fungible-token(gFT). This is necessary to achieve idempotency and state synchronization.
 
-The default implementation of the NFT contract is provided in the Gear library: [gear-lib/multitoken](https://github.com/gear-dapps/gear-lib/tree/master/lib/src/multitoken).
+### The storage contract(mt-storage)
 
-To use the default implementation you should include the packages into your Cargo.toml file:
-```toml
-gear-contract-libraries = { path = "https://github.com/gear-dapps/gear-lib" }
-derive_traits = { path = "https://github.com/gear-dapps/gear-lib/tree/master/derive" }
-hashbrown = "0.13.1"
-```
-
-The states that multitoken contract store are defined in the struct MTKState:
-
-```rust
-use hashbrown::HashMap;
-
-#[derive(Debug, Default)]
-pub struct MTKState {
-    pub name: String,
-    pub symbol: String,
-    pub base_uri: String,
-    pub balances: HashMap<TokenId, HashMap<ActorId, u128>>,
-    pub approvals: HashMap<ActorId, HashMap<ActorId, bool>>,
-    pub token_metadata: HashMap<TokenId, TokenMetadata>,
-    // owner for nft
-    pub owners: HashMap<TokenId, ActorId>,
-}
-```
+Saves the user's balances and approvals. Contains commands to manipulate these fields. When the user interacts with `mt-logic`, a new associated `mt-storage` is created - this provides scalability and greatly increases memory limits.
```

Comment:
`mt-storage` isn't created every time but from the description it seems when every time when user  interacts with `mt-logic`, a new `mt-storage` is created .